packages/contracts-bedrock/contracts/test/StandardBridge.t.sol

```diff
@@ -0,0 +1,134 @@
+import { CommonTest } from "./CommonTest.t.sol";
+import {
+    OptimismMintableERC20,
+    ILegacyMintableERC20
+} from "../universal/OptimismMintableERC20.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/**
+ * @title StandardBridgeTester
+ * @notice Simple wrapper around the StandardBridge contract that exposes
+ *         internal functions so they can be more easily tested directly.
+ */
+contract StandardBridgeTester is StandardBridge {
+    constructor(address payable _messenger, address payable _otherBridge)
+        StandardBridge(_messenger, _otherBridge)
+    {}
+
+    function isOptimismMintableERC20(address _token) external view returns (bool) {
+        return _isOptimismMintableERC20(_token);
+    }
+
+    function isCorrectTokenPair(address _mintableToken, address _otherToken)
+        external
+        view
+        returns (bool)
+    {
+        return _isCorrectTokenPair(_mintableToken, _otherToken);
+    }
+
+    receive() external payable override {}
+}
+
+/**
+ * @title LegacyMintable
+ * @notice Simple implementation of the legacy OptimismMintableERC20.
+ */
+contract LegacyMintable is ERC20, ILegacyMintableERC20 {
+    constructor(string memory _name, string memory _ticker) ERC20(_name, _ticker) {}
+
+    function l1Token() external view returns (address) {
+        return address(0);
+    }
+
+    function mint(address _to, uint256 _amount) external pure {}
+
+    function burn(address _from, uint256 _amount) external pure {}
+
+    /**
+     * @notice Implements ERC165. This implementation should not be changed as
+     *         it is how the actual legacy optimism mintable token does the
+     *         check. Allows for testing against code that is has been deployed,
+     *         assuming different compiler version is no problem.
+     */
+    function supportsInterface(bytes4 _interfaceId) external view returns (bool) {
+        bytes4 firstSupportedInterface = bytes4(keccak256("supportsInterface(bytes4)")); // ERC165
+        bytes4 secondSupportedInterface = ILegacyMintableERC20.l1Token.selector ^
+            ILegacyMintableERC20.mint.selector ^
+            ILegacyMintableERC20.burn.selector;
+        return _interfaceId == firstSupportedInterface || _interfaceId == secondSupportedInterface;
+    }
+}
+
+/**
+ * @title StandardBridge_Stateless_Test
+ * @notice Tests internal functions that require no existing state or contract
+ *         interactions with the messenger.
+ */
+contract StandardBridge_Stateless_Test is CommonTest {
+    StandardBridgeTester internal bridge;
+    OptimismMintableERC20 internal mintable;
+    ERC20 internal erc20;
+    LegacyMintable internal legacy;
+
+    function setUp() external {
+        bridge = new StandardBridgeTester({
+            _messenger: payable(address(0)),
+            _otherBridge: payable(address(0))
+        });
+
+        mintable = new OptimismMintableERC20({
```

Comment:
One thing here is that [OptimismMintableERC20](https://github.com/maurelian/optimism/blob/3261f54860bb91bcd0edc3992c1f2f0a8261142e/packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol#L17) inherits from the Legacy token as well, which I think is one of major reasons we missed this compatibility issue, and I worry we're still not testing that properly. 

Could we find a way to make sure that all the StandardBridge tests are run on both tokens?

Reply:
Is there any reason we can't just remove `ILegacyMintableERC20` from `OptimismMintableERC20`?

Reply:
> Could we find a way to make sure that all the StandardBridge tests are run on both tokens?

I'd prefer to scope this in another task as its a refactor of our tests if thats cool. We need to dedupe the L1 and L2 tests while we are at it. I'd like this PR to be focused on unit tests around the tested functions

Reply:
> Is there any reason we can't just remove ILegacyMintableERC20 from OptimismMintableERC20?

This may cause a headache for offchain tooling, they would need to do the interface checking themselves offchain. I'm not super opposed to it to be honest tho, perhaps we should think about the impacts of the decision before making it

Reply:
> I'd prefer to scope this in another task as its a refactor of our tests if thats cool. 

I do think that in order to be confident that these issues are fully resolved we need to fully test tokens which are exclusively `IOptimismMintableERC20` and `ILegacyMintableERC20`. It's fine with me if that happens in another PR/ticket.